examples/quantization_aware_training/torch/resnet18/main.py

```diff
@@ -0,0 +1,325 @@
+
+    print(f"Accuracy of tuned INT8 model: {acc1_int8:.3f}")
+    print(f"Accuracy drop of tuned INT8 model over pre-trained FP32 model: {acc1_fp32 - acc1_int8:.3f}")
+
+    input_shape = (1, 3, IMAGE_SIZE, IMAGE_SIZE)
+    example_input = torch.randn(*input_shape).cpu()
+
+    # Export FP32 model to OpenVINO™ IR
+    fp32_ir_path = f"{ROOT}/{BASE_MODEL_NAME}_fp32.xml"
+    ov_model = ov.convert_model(model.cpu(), example_input=example_input)
+    ov.save_model(ov_model, fp32_ir_path, compress_to_fp16=False)
+    print(f"FP32 model was exported to {fp32_ir_path}.")
+
+    # Export INT8 model to OpenVINO™ IR
+    int8_ir_path = f"{ROOT}/{BASE_MODEL_NAME}_int8.xml"
+    warnings.filterwarnings("ignore", category=TracerWarning)
+    warnings.filterwarnings("ignore", category=UserWarning)
+    ov_model = ov.convert_model(quantized_model.cpu(), example_input=example_input)
+    ov.save_model(ov_model, int8_ir_path)
+    print(f"INT8 model exported to {int8_ir_path}.")
+
+    print("Benchmark FP32 model (IR)")
+    fp32_fps = run_benchmark(fp32_ir_path, shape=input_shape, verbose=True)
+
+    print("Benchmark INT8 model (IR)")
+    int8_fps = run_benchmark(int8_ir_path, shape=input_shape, verbose=True)
+
+    print(f"Performance speed up (throughput mode): {int8_fps / fp32_fps:.3f}")
+
+    fp32_model_size = get_model_size(fp32_ir_path, verbose=True)
+    int8_model_size = get_model_size(int8_ir_path, verbose=True)
+    print(f"Model compression rate: {fp32_model_size / int8_model_size:.3f}")
+    return acc1_fp32, acc1_int8_init, acc1_int8, fp32_fps, int8_fps, fp32_model_size, int8_model_size
+
+
+def train(
+    train_loader: torch.utils.data.DataLoader,
+    model: torch.nn.Module,
+    criterion: torch.nn.Module,
+    optimizer: torch.optim.Optimizer,
+    epoch: int,
+    device: torch.device,
+):
+    print("Training:")
+    # Switch to train mode.
+    model.train()
+
+    print_frequency = 50
+    running_loss = 0.0
+    for i, (images, target) in enumerate(train_loader):
+        images = images.to(device)
+        target = target.to(device)
+
+        # Compute output.
+        output = model(images)
+        loss = criterion(output, target)
+
+        # Compute gradient and do opt step.
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+
+        running_loss += loss.item()
+        if i % print_frequency == print_frequency - 1:
+            print(f"[Epoch {epoch + 1}, step {i + 1:5d}] loss: {running_loss / print_frequency:.3f}")
+            running_loss = 0.0
+
+
+def validate(
+    val_loader: torch.utils.data.DataLoader, model: torch.nn.Module, criterion: torch.nn.Module, device: torch.device
+) -> float:
+    print("Validation:")
+    running_loss = 0.0
+    top1_sum = running_top1 = 0.0
+    top5_sum = running_top5 = 0.0
+    print_frequency = 10
+
+    # Switch to evaluate mode.
+    model.eval()
+
+    with torch.no_grad():
+        for i, (images, target) in enumerate(val_loader):
```

Comment:
To reduce log suggest ot use 
```
    for i, (images, target) in track(enumerate(val_loader), total=len(val_loader), description="Validation"):
```
and remove extra this print, and do same for training

Reply:
I suggest to keep it as is, progress printing + track looks strange

Reply:
I have suggested to keep only use track without any other outputs.

Is it really need to present QAT show to user accuracy and loss on first 50, 100, 150 ... images? I sure that no. 
```
[Step 9] acc1 62.109375, acc5 84.296875, loss 1.58587127327919
[Step 19] acc1 54.0625, acc5 81.328125, loss 1.8110883593559266
[Step 29] acc1 54.6875, acc5 78.59375, loss 1.9364415168762208
[Step 39] acc1 51.328125, acc5 77.5, loss 2.01123251914978
[Step 49] acc1 56.5625, acc5 79.6875, loss 1.8059288382530212
[Step 59] acc1 53.515625, acc5 77.578125, loss 1.9870043992996216
[Step 69] acc1 52.890625, acc5 75.9375, loss 2.035608100891113
````

Same for training
```
[Epoch 1, step    50] loss: 0.811
[Epoch 1, step   100] loss: 0.798
[Epoch 1, step   150] loss: 0.779
[Epoch 1, step   200] loss: 0.752
[Epoch 1, step   250] loss: 0.763
[Epoch 1, step   300] loss: 0.740
[Epoch 1, step   350] loss: 0.769
[Epoch 1, step   400] loss: 0.744
[Epoch 1, step   450] loss: 0.764
[Epoch 1, step   500] loss: 0.730
[Epoch 1, step   550] loss: 0.721
```

Reply:
My reference was this example https://github.com/openvinotoolkit/nncf/pull/2515#pullrequestreview-1899673069, if you think nobody need this I'll remove it

Reply:
Done